components/planner/history/StudentHistoryView/CourseAudit.tsx

```diff
@@ -0,0 +1,686 @@
+  order: Order;
+  orderBy: string;
+  rowCount: number;
+}
+
+//implemented tablehead
+function EnhancedTableHead({
+  classes,
+  onSelectAllClick,
+  order,
+  orderBy,
+  numSelected,
+  rowCount,
+  onRequestSort,
+}: EnhancedTableProps) {
+  const createSortHandler = (property: string) => (event: React.MouseEvent<unknown>) => {
+    onRequestSort(event, property);
+  };
+
+  return (
+    <TableHead>
+      <TableRow>
+        <TableCell padding="checkbox"></TableCell>
+        {headCells.map((headCell) => (
+          <TableCell
+            key={headCell.id}
+            align={headCell.numeric ? 'left' : 'left'}
+            padding={headCell.disablePadding ? 'none' : 'default'}
+            sortDirection={orderBy === headCell.id ? order : false}
+          >
+            <TableSortLabel
+              active={orderBy === headCell.id}
+              direction={orderBy === headCell.id ? order : 'asc'}
+              onClick={createSortHandler(headCell.id)}
+            >
+              {headCell.label}
+              {orderBy === headCell.id ? (
+                <div className={classes.visuallyHidden}>
+                  {order === 'desc' ? 'sorted descending' : 'sorted ascending'}
+                </div>
+              ) : null}
+            </TableSortLabel>
+          </TableCell>
+        ))}
+      </TableRow>
+    </TableHead>
+  );
+}
+
+//stylesheet
+const useToolbarStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      paddingLeft: theme.spacing(2),
+      paddingRight: theme.spacing(1),
+    },
+    highlight:
+      theme.palette.type === 'light'
+        ? {
+            color: theme.palette.secondary.main,
+            backgroundColor: lighten(theme.palette.secondary.light, 0.85),
+          }
+        : {
+            color: theme.palette.text.primary,
+            backgroundColor: theme.palette.secondary.dark,
+          },
+    title: {
+      flex: '1 1 100%',
+    },
+    formControl: {
+      margin: theme.spacing(1),
+      minWidth: 120,
+    },
+    selectEmpty: {
+      marginTop: theme.spacing(2),
+    },
+    wrapper: {
+      display: 'flex',
+      justifyContent: 'space-between',
+      flexWrap: 'wrap',
+      width: '100%',
+    },
+    wider: {
+      width: '10%',
+    },
+    course: {
+      paddingTop: '22px',
+      width: '20%',
+    },
+  }),
+);
+
+/**
+ * Allows adding and deleting courses
+ * numSelected: the number of rows selected
+ * data: the array of CourseAttempt
+ * selectedState: the list of course names that have been selected
+ * update(CourseAttempt[]): updates the rows
+ * updateNum(string[]): updates the number selected
+ * updateSelectedState(string[]): updates the list of courses selected
+ */
+interface EnhancedTableToolbarProps {
+  numSelected: number;
+  data: CourseAttempt[];
+  selectedState: string[];
+  update(courses: CourseAttempt[]): any;
+  updateNum(courses: string[]): any;
+  updateSelectedState(courses: string[]): any;
```

Comment:
This is how function types are supposed to look:
```suggestion
  update: (courses: CourseAttempt[]) => any;
  updateNum: (courses: string[]) => any;
  updateSelectedState: (courses: string[]) => any;
```
Additionally, you should not be using `any` types unless you're certain they're necessary. What is the most restrictive variable type that you can use with these functions?